src/Player.cpp

```diff
@@ -51,27 +61,35 @@ void Player::SaveGame(){
 void Player::SetPlayerData(){
     // Primarily initializes default values at the beginning of the game.
 
-	ifstream ReadData;
-	ReadData.clear();
-	ReadData.open("data.txt");
-	if (ReadData.is_open())	{
-		ReadData >> player_type;
-    ReadData.ignore();       // Ignore rest of line ready for getline
-    getline(ReadData, name);
-    ReadData >> gender;
-    ReadData >> level;
-    ReadData >> experience;
-		ReadData >> health;
-		ReadData >> arrows;
-		ReadData >> bombs;
-		ReadData >> potions;
-		ReadData >> whetstones;
-		ReadData >> weaponstrength;
-		ReadData >> coins;
+	std::ifstream ReadData("data.json");
+	if (ReadData && ReadData.peek() != EOF)
+	{
+		json SaveDataJson = json::parse(ReadData);
+		if (SaveDataJson.size() != 12)
+		{
+			cout << "savegame data length mismatch (data.json)." << endl;
+			return;
```

Comment:
What if the number of attributes changes? Instead of checking the number of entries in the JSON file, we probably want to just check that each of the keys we're looking for exists in the JSON file. Also, if we just return here, the game still continues without actually setting player data - we probably want to go back to the main menu.